frame/uniswap-v2/src/lib.rs

```diff
@@ -0,0 +1,863 @@
+//
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![warn(
+	bad_style,
+	bare_trait_objects,
+	const_err,
+	improper_ctypes,
+	non_shorthand_field_patterns,
+	no_mangle_generic_items,
+	overflowing_literals,
+	path_statements,
+	patterns_in_fns_without_body,
+	private_in_public,
+	unconditional_recursion,
+	unused_allocation,
+	unused_comparisons,
+	unused_parens,
+	while_true,
+	trivial_casts,
+	trivial_numeric_casts,
+	unused_extern_crates
+)]
+#![allow(clippy::all)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::{Codec, FullCodec};
+	use composable_traits::{
+		currency::CurrencyFactory,
+		dex::{ConstantProductPoolInfo, CurveAmm, PoolData, PoolId, PoolTokenIndex},
+		math::LiftedFixedBalance,
+	};
+	use frame_support::{
+		pallet_prelude::*,
+		traits::fungibles::{Inspect, Mutate, Transfer},
+		PalletId,
+	};
+	use scale_info::TypeInfo;
+	use sp_core::crypto::KeyTypeId;
+	use sp_runtime::{
+		traits::{
+			AccountIdConversion, AtLeast32BitUnsigned, CheckedAdd, CheckedDiv, CheckedMul,
+			CheckedSub, IntegerSquareRoot, One, Zero,
+		},
+		FixedPointNumber, FixedPointOperand, FixedU128, KeyTypeId as CryptoKeyTypeId, Permill,
+	};
+	use sp_std::{collections::btree_set::BTreeSet, fmt::Debug, iter::FromIterator};
+
+	pub const PALLET_ID: PalletId = PalletId(*b"Uni-swap");
+	pub const KEY_TYPE: KeyTypeId = KeyTypeId(*b"uswp");
+	pub const CRYPTO_KEY_TYPE: CryptoKeyTypeId = CryptoKeyTypeId(*b"uswp");
+
+	pub mod crypto {
```

Comment:
not sure, is it used somewhere or will be?

Reply:
I hope it will be used. The current code is not complete pallet as it still requires extrinsics. It may need OCW. if not used in final version of pallet we can remove these.

Reply:
From what I understand, the uniswap pallet don't need any OCW, we can remove this.